network/onchain-discovery/src/service.rs

```diff
@@ -131,13 +129,21 @@ async fn handle_query_discovery_set_request(
     req_msg: QueryDiscoverySetRequest,
     mut res_tx: oneshot::Sender<Result<Bytes, RpcError>>,
 ) {
+    let mut f_query_storage =
+        task::spawn_blocking(move || storage_query_discovery_set(libra_db, req_msg)).fuse();
```

Comment:
just looked this up... recommend that storage_query_discovery_set have the task::spawn_blocking internally that way you can have common handling logic.